atef/widgets/config/window.py

```diff
@@ -556,9 +553,8 @@ def refresh_model(self) -> None:
 
         # Clear widget caches
         for cache in (self.edit_widget_cache, self.run_widget_cache):
-            for widget in cache.values():
-                widget.setParent(None)
-                widget.deleteLater()
+            for _ in range(len(cache)):
+                cache.popitem()
```

Comment:
Is this a continuation of the "let python do the garbo collection" mantra?

Reply:
yes it is.  Ran into a similar problem and applied the same solution